eeprom/tests/test_serial_number.cpp

```diff
@@ -0,0 +1,100 @@
+#include <vector>
+
+#include "catch2/catch.hpp"
+#include "eeprom/core/serial_number.hpp"
+
+using namespace eeprom;
+
+struct MockEEPromTaskClient {
+    void send_eeprom_queue(const task::TaskMessage& m) {
```

Comment:
We actually already have a shared mock queue client if you want to use it. It's in `include/common/tests`